File: POS_Front_End/js/payment.js
$(document).ready(function () {
    loadPayments();
});

function loadPayments() {

    $.ajax({
        type: "GET",
        url: "http://localhost:8080/api/v1/payment",

        success: function (response) {

            $("#payment-table tbody").empty();

            $.each(response.data, function (index, payment) {

                let row = `
                    <tr>
                        <td>${payment.paymentId}</td>
                        <td>${payment.orderId}</td>
                        <td>${payment.customerName}</td>
                        <td>${payment.itemNames}</td>
                        <td>Rs. ${payment.amount.toFixed(2)}</td>
                        <td>${payment.paymentMethod}</td>
                        <td>${payment.dateTime.replace("T"," ")}</td>
                    </tr>
                `;

                $("#payment-table tbody").append(row);
            });
        }
    });
}

File: POS_Back_End/src/main/java/lk/ijse/posbackend/entity/Payment.java
package lk.ijse.posbackend.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Payment {

    @Id
    private String paymentId;

    private String orderId;

    private String customerName;

    @Column(length = 500)
    private String itemNames;

    private double amount;

    private String paymentMethod;

    private LocalDateTime dateTime;
}